trin-core/src/cli.rs

```diff
@@ -172,18 +212,17 @@ impl TrinConfig {
         I: Iterator<Item = T>,
         T: Into<OsString> + Clone,
     {
-        let config = Self::from_iter(args);
+        let config = Self::from_iter_safe(args).unwrap_or_else(|e| panic!("{e}"));
```

Comment:
Why are we changing this from [from_iter](https://docs.rs/structopt/latest/structopt/trait.StructOpt.html#method.from_iter)?

Also `unwrap_or_else` and then panicking doesn't make sense to me, isn't the same as just `.unwrap()`?

Reply:
I think this is a way to explicitly state that we have handled the `unwrap` here and are using the default error message instead of a custom `expect` message ?

It was suggested by @njgheorghita

Reply:
This introduced a panic during `--help` or `--version`.See https://docs.rs/structopt/0.3.7/structopt/trait.StructOpt.html#method.from_iter_safe

New test to detect the panic, plus the fix, in #617